Support for Custom Fields

CristhianMotoche opened the pull request:
Hello!
Sorry for the taking some time with this PR.

## Notes
1) I updated the `circle.yml` file to the latest version
2) I've been using [poetry](https://poetry.eustace.io/) lately and I consider using it in the project to facilitate isolation and reproducibility. However, if you think this tool is not needed I can remove it to work without it.
3) Unfortunately, I couldn't recreate VCR cassettes with customFields for my tests, so I just copied one of yours and set a specific content for the customFields, as it is specified in [hatchbucks api](https://hatchbuck.freshdesk.com/support/solutions/articles/5000578765-hatchbuck-api-documentation-for-advanced-users-) documentation.

Solves #18

CristhianMotoche committed:
Parse name and type of customFields

CristhianMotoche committed:
Add poetry to run the project

CristhianMotoche committed:
Add test when there are no customFields

CristhianMotoche committed:
Merge pull request #1 from CristhianMotoche/parse_customfields_accordingly_api

Parse customfields accordingly api

jakesen reviewed (commented):
Awesome! Everything looks good to me. Do you think we should add documentation in the readme for using the custom fields and for setting up the development/test environment with poetry?

CristhianMotoche committed:
Add instructions to add customFields

CristhianMotoche commented:
I updated the README with instructions to set the project with poetry, add Custom Fields using a new method called `add_customField` and I added a tests for it. Let me know what you think. :slightly_smiling_face:

jakesen commented on hatchbuck/objects.py:
```diff
@@ -109,6 +108,10 @@ def __init__(self, api_key, json_data=None):
     def add_email(self, address, type):
         self.emails.append(Email({'address': address, 'type': type}))
 
+    def add_customField(self, name, value):
```
For consistency with other methods in this class, we should use snake case for the method name: `add_custom_field`

CristhianMotoche replied:
Sure! That make sense.

jakesen reviewed (changes requested):
Updates to the readme look great! I think if we can change that function name, we will be good to go!

CristhianMotoche committed:
Rename add_customFields to add_custom_fields for consistency

jakesen reviewed (approved):
@CristhianMotoche: Thanks for your contribution to pyhatchbuck!

CristhianMotoche commented:
It was a pleasure! I hope you can update this in the [PyPi repository](https://pypi.org/project/pyhatchbuck/#history) any time soon. :slightly_smiling_face:

jakesen commented:
> It was a pleasure! I hope you can update this in the [PyPi repository](https://pypi.org/project/pyhatchbuck/#history) any time soon.

Just published under version 0.2 :-)

I also tested the PyPi hosted package in one of my projects and everything seems to work fine.
